MovieGPT-api/src/main/java/lt/techin/group/project/model/User.java

```diff
@@ -0,0 +1,28 @@
+package lt.techin.group.project.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import jakarta.persistence.*;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.Set;
+
+@Getter
+@Setter
+@Entity
+@Table(name = "users")
+public class User {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private String username;
+    private String email;
+    @JsonIgnore
+    private String password;
+
+    @ManyToMany(fetch = FetchType.EAGER)
```

Comment:
My idea behind this List of Roles in User, so like admin must have all roles, basic user usually have one role like "user".
So when we going to add "authority" to endpoint, example:
.requestMatchers(HttpMethod.GET, "/api/movies").hasAnyAuthority( _**USER**_ )

We don't need to add "ADMIN" role everywhere, because admin will have "user" role and can access every point where basic user can go.

And we can secure endpoint for admin only when we need it:

.requestMatchers(HttpMethod.DELETE, "/api/movies/{id}").hasAnyAuthority( _**ADMIN**_ )